### ISSUE TITLE
Program ignores edge case with CAPTCHA response from server

### ISSUE THREAD

**Issue by phanhoang1366** [atomicptr/hoyo-daily-logins-helper#11]
**Issue Description:**

The current `game_perform_checkin()` function only checks if `retcode` is equal to 0 or not. My account seems to get in a difficult situation: I have to complete a CAPTCHA from the browser in order to check in. This creates additional data from `data: gt_request`, while also sets `retcode` to 0. This makes the program believes that it has checked in correctly, while it's not true.

**Reproduce:**

This is hard to reproduce, since it's usually random and differs from account to account.

**Environment:**

- Python version: Python 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59)
- Operating System: Windows 10

**POST response sample:**

With CAPTCHA check:
```json
{
  "retcode": 0,
  "message": "OK",
  "data": {
    "code": "ok",
    "first_bind": false,
    "gt_result": {
      "risk_code": 5001,
      "gt": "a280f1e5745a37ae91c65fc745fba71c",
      "challenge": "39d34da9de2c4f6bac4489990dab3c0b",
      "success": 1,
      "is_risk": true
    }
  }
}
```

After completing CAPTCHA check:
```json
{
  "retcode": 0,
  "message": "OK",
  "data": {
    "code": "ok",
    "first_bind": false,
    "gt_result": {
      "risk_code": 0,
      "gt": "",
      "challenge": "",
      "success": 0,
      "is_risk": false
    }
  }
}
```

**Comment by atomicptr** [comment]
Interesting, I have been running a version of this for Genshin for years and never had a captcha block me 😂. Did you by chance edit the user agent?

I'll def look into this, perhaps send an error like "please complete captcha" or something to the chat 🤔.

**Comment by phanhoang1366** [comment]
I will try it next time I check-in with a different user-agent. I just used the default one that came with the program.

The check-in page didn't even throw an error, it just behaves like when you login to Genshin or Hoyolab (automatically shows the CAPTCHA page without an error of some kind).

**Comment by atomicptr** [comment]
Then that's probably not the issue, I noticed a lot of people use the example one from the Readme which is might trigger something like this. Interesting either way 😂

**Comment by atomicptr** [comment]
This should be solved with v2.5.2, which will send an error message with a link to the login page when the login fails. Since I can't really test this (or probably anyone else) I'll close this issue as completed but should this issue re-occur feel free to re-open the ticket! Thanks for reporting!

**Comment by atomicptr** [comment]
Got a response from someone else that makes it seem like they are working on some preventative measures.

![image](https://github.com/atomicptr/hoyo-daily-logins-helper/assets/674705/4e689cb3-afb1-4cf5-9cef-f5a62307bb68)

**Comment by atomicptr** [comment]
They haven't deleted the gt_result thing completely as the parameters still appear on an account, so I added a super generic check to see if the user actually signed in after the automatic attempt and if not send an error message in Discord! They probably can't break this unless they change this endpoint significantly. Fixed with v2.5.3
